n3fit/src/n3fit/backends/keras_backend/MetaModel.py

```diff
@@ -378,10 +377,10 @@ def set_replica_weights(self, weights, i_replica=0):
             i_replica: int
                 the replica number to set, defaulting to 0
         """
-        self.get_layer(f"{NN_PREFIX}_{i_replica}").set_weights(weights[NN_PREFIX])
-        self.get_layer(f"{PREPROCESSING_PREFIX}_{i_replica}").set_weights(
-            weights[PREPROCESSING_PREFIX]
-        )
+        for layer_type in [NN_LAYER, PREPROESSING_LAYER]:
+            set_layer_replica_weights(
+                layer=self.get_layer(layer_type), weights=weights[layer_type], i_replica=i_replica
+            )
```

Comment:
```suggestion
        for l_type in [NN_LAYER, PREPROESSING_LAYER]:
            layer = self.get_layer(l_type)
            set_layer_replica_weights(layer=layer, weights=weights[l_type], i_replica=i_replica)
```

Yes, yes, feel free to ignore this one while sighing

Reply:
Haha no I agree it's nicer, fixed :)